src/services/conversationsService.js

```diff
@@ -0,0 +1,28 @@
+import api from "../api/api";
+import store from "../store/store";
+import { insertChats } from "../store/Conversations";
+
+class ConversationsService {
```

Comment:
I what if we subscribe inside constructor. Any concerns re it?

e/g/


```
import api from "../api/api";
import store from "../store/store";
import { insertChats } from "../store/Conversations";

class ConversationsService {
  userIsLoggedIn = false;

  constructor() {
    store.subscribe(() => {
      let previousValue = this.userIsLoggedIn;
      this.userIsLoggedIn = store.getState().userAuth.value;
    
      if (this.userIsLoggedIn && !previousValue) {
        conversationService.syncData();
      }
    });
  }

  async syncData() {
    api
      .conversationList({})
      .then((chats) => store.dispatch(insertChats(chats)));
  }
}

const conversationService = new ConversationsService();

export default conversationService;
```

Looks a bit nicer fron my perspective. WDYT?